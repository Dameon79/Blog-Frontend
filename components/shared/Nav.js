import { Component } from 'react'

class Nav extends Component {
  constructor (props) {
    super(props)

    this.state = {
      burgerNav: "c-burger-container"
    }

    this.animateBurger = this.animateBurger.bind(this)
  }

  animateBurger () {
    const { burgerNav } = this.state

    if (burgerNav === 'c-burger-container') {
      this.setState({
        burgerNav: "c-burger-container change"
      })  
    } else {
      this.setState({
        burgerNav: "c-burger-container"
      })
    }
  }

  render () {
    const { burgerNav } = this.state

    return (
      <div className={burgerNav} onClick={this.animateBurger}>
        <div className="c-burger-container__burger1"></div>
        <div className="c-burger-container__burger2"></div>
        <div className="c-burger-container__burger3"></div>
    
    
        <nav className="c-burgernav">
          <a href="#" className="c-burgernav__menu">Home</a>
          <a href="#" className="c-burgernav__menu">About Me</a> 
          <a href="#" className="c-burgernav__menu">My Projects</a> 
          <a href="#" className="c-burgernav__menu">Blog Articles</a> 
          <a href="#" className="c-burgernav__menu u-white-border">Contact</a>  
        </nav>
      </div>
    );
  }
}

export default Nav