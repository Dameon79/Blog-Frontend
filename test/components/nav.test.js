import Nav from '../../components/shared/Nav'

describe('<Nav />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Nav />
    )

    expect(component).toMatchSnapshot()
  })
})