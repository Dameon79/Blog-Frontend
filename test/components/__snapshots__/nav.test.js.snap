// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-burger-container__burger1"
        />,
        <div
          className="c-burger-container__burger2"
        />,
        <div
          className="c-burger-container__burger3"
        />,
        <nav
          className="c-burgernav"
        >
          <Link
            href="/"
          >
            <a
              className="c-burgernav__menu"
            >
              Home
            </a>
          </Link>
          <Link
            href="/about"
          >
            <a
              className="c-burgernav__menu"
            >
              About Me
            </a>
          </Link>
          <Link
            href="/projects"
          >
            <a
              className="c-burgernav__menu"
            >
              My Projects
            </a>
          </Link>
          <Link
            href="/articles"
          >
            <a
              className="c-burgernav__menu"
            >
              Blog Articles
            </a>
          </Link>
          <Link
            href="/contact"
          >
            <a
              className="c-burgernav__menu u-white-border"
            >
              Contact
            </a>
          </Link>
        </nav>,
      ],
      "className": "c-burger-container",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-burger-container__burger1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-burger-container__burger2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-burger-container__burger3",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href="/"
            >
              <a
                className="c-burgernav__menu"
              >
                Home
              </a>
            </Link>,
            <Link
              href="/about"
            >
              <a
                className="c-burgernav__menu"
              >
                About Me
              </a>
            </Link>,
            <Link
              href="/projects"
            >
              <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>
            </Link>,
            <Link
              href="/articles"
            >
              <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>
            </Link>,
            <Link
              href="/contact"
            >
              <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>
            </Link>,
          ],
          "className": "c-burgernav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                Home
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Home",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "Home",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                About Me
              </a>,
              "href": "/about",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "About Me",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "About Me",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>,
              "href": "/projects",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Projects",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "My Projects",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>,
              "href": "/articles",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Blog Articles",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "Blog Articles",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>,
              "href": "/contact",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact",
                "className": "c-burgernav__menu u-white-border",
              },
              "ref": null,
              "rendered": "Contact",
              "type": "a",
            },
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-burger-container__burger1"
          />,
          <div
            className="c-burger-container__burger2"
          />,
          <div
            className="c-burger-container__burger3"
          />,
          <nav
            className="c-burgernav"
          >
            <Link
              href="/"
            >
              <a
                className="c-burgernav__menu"
              >
                Home
              </a>
            </Link>
            <Link
              href="/about"
            >
              <a
                className="c-burgernav__menu"
              >
                About Me
              </a>
            </Link>
            <Link
              href="/projects"
            >
              <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>
            </Link>
            <Link
              href="/articles"
            >
              <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>
            </Link>
            <Link
              href="/contact"
            >
              <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>
            </Link>
          </nav>,
        ],
        "className": "c-burger-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-burger-container__burger1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-burger-container__burger2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-burger-container__burger3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href="/"
              >
                <a
                  className="c-burgernav__menu"
                >
                  Home
                </a>
              </Link>,
              <Link
                href="/about"
              >
                <a
                  className="c-burgernav__menu"
                >
                  About Me
                </a>
              </Link>,
              <Link
                href="/projects"
              >
                <a
                  className="c-burgernav__menu"
                >
                  My Projects
                </a>
              </Link>,
              <Link
                href="/articles"
              >
                <a
                  className="c-burgernav__menu"
                >
                  Blog Articles
                </a>
              </Link>,
              <Link
                href="/contact"
              >
                <a
                  className="c-burgernav__menu u-white-border"
                >
                  Contact
                </a>
              </Link>,
            ],
            "className": "c-burgernav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  Home
                </a>,
                "href": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "Home",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  About Me
                </a>,
                "href": "/about",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "About Me",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "About Me",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  My Projects
                </a>,
                "href": "/projects",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Projects",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "My Projects",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  Blog Articles
                </a>,
                "href": "/articles",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Blog Articles",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "Blog Articles",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu u-white-border"
                >
                  Contact
                </a>,
                "href": "/contact",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contact",
                  "className": "c-burgernav__menu u-white-border",
                },
                "ref": null,
                "rendered": "Contact",
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
